collections/index.js

```diff
@@ -20,46 +20,130 @@ import employees from './data.json';
   go for it - that's the whole notion of functional programming
 
   The format for this test will be slightly different - the data that I will
-  be using for the test suite lives in ./data.json. I have written you an 
+  be using for the test suite lives in ./data.json. I have written you an
   import statement and you should use these data sets for the test.
 */
+export const howManyEmployees = () => {
+  return employees.length;
+}
 
 /*
   I expect this function to get me an employee by their ID
 */
 export const getEmployeeById = (id) => {
-
+  // created an empty object to insert data into
+  var employeeNumber = {};
+  // using a for loop to index through each of the employees on record
+  for (let employeeNumberIndex = 0; employeeNumberIndex < employees.length; employeeNumberIndex++) {
+    // in each of the iterations, if the employee property 'id' matches the
+    // provided arguement 'id', the statement is true...
+    if (employees[employeeNumberIndex].id === id) {
+      // ... and the following code is executed:
+      // this assigns the employees details matching their id to the
+      // variable 'employeeNumber'
+      employeeNumber = employees[employeeNumberIndex];
+    }
+  }
+  // this returns the employee details as the result of the function.
+  return employeeNumber;
 }
 
 /*
   I expect this function to return me an array of employees depending on which
   department is passed in
 */
 export const getEmployeesByDepartment = (dept) => {
-
+  // sets an empty array ready to accept any values we add later:
+  var departmentEmployees = [];
+  // using a for loop to index through each of the employees on record
+  for (let employeeNumberIndex = 0; employeeNumberIndex < employees.length; employeeNumberIndex++) {
```

Comment:
Same as above regarding naming the loop index `i` as opposed to `employeeNumberIndex`

Reply:
done